src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +87,13 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<People> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(person -> person.getSex().equals(People.Sex.WOMEN)
+                        && person.getAge() >= femaleAge)
+                .map(People::getCats)
+                .flatMap(Collection::stream)
```

Comment:
Optional. But you can make it in one line too.
```suggestion
                .flatMap(p -> p.getCats().stream())
```